### PR TITLE
Refactor code related to issue models 

### PR DESCRIPTION
**Pull request by eduardoj**
We realized that the issues' models had unused code. In order to have a better codebase we decided to cleanup the unused code and refactor.

Have a look at the commits' descriptions for a more detailed explanation.

Co-authored-by: David Kang <dkang@suse.com>
Co-authored-by: Saray Cabrera Padrón <scabrerapadron@suse.de>

### CODE REVIEW

File: src/api/app/models/issue.rb [hunk 1]
```diff
@@ -4,6 +4,7 @@ class Issue < ApplicationRecord
   class NotFoundError < APIError
     setup 'issue_not_found', 404, 'Issue not found'
   end
+  class InvalidName < APIError; end
```
Comment (vpereira) [inline src/api/app/models/issue.rb]: please move all the error classes to `app/models/issue/errors.rb` as we have for the others?

Comment (coolo) [reply src/api/app/models/issue.rb]: a concern for 4 lines? Isn't that a little over the top?

Comment (vpereira) [reply src/api/app/models/issue.rb]: all errors classes. Can be more in the future. It's matter of code organization and structure.

Comment (eduardoj) [reply src/api/app/models/issue.rb]: Done. Issue errors were moved to its own class. Thanks!

**Pull request merged**